internal/repos/modules_test.go

```diff
@@ -36,3 +43,22 @@ func TestGoModuleSpecialCases(t *testing.T) {
 		})
 	}
 }
+
+func TestGoModuleReplacement(t *testing.T) {
```

Comment:
I'd prefer not to test against `toRepoRule` directly, since it's an unexported function and we should be able to restructure its functionality without breaking tests. In any case, this seems redundant with import_test.go.